cognitive/src/main/scala/com/microsoft/azure/synapse/ml/services/openai/GlobalParam.scala

```diff
@@ -0,0 +1,161 @@
+package com.microsoft.azure.synapse.ml.services.openai
+
+import com.microsoft.azure.synapse.ml.param.ServiceParam
+import com.microsoft.azure.synapse.ml.services.HasServiceParams
+import org.apache.spark.ml.param.Param
+import org.apache.spark.sql.Row
+
+import scala.collection.mutable
+
+sealed trait GlobalParamKey[T]
+case object OpenAIDeploymentNameKey extends GlobalParamKey[String]
```

Comment:
We should have the ability to poerate this in an untyped way as well as a typed way. Because to go from python into scala we'll only have strings

Reply:
Added explicit setters for each of the keys (i.e. setDeploymentName). We'll use this in both Scala and Python.